plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend

```diff
@@ -153,24 +149,34 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 	 */
 	@Check
 	def checkWithExecutionHasDeactivate(Model model) {
-		var messagesWithExecution = newHashMap
+		// keep a list with the messages that contains the withExecution keyword
+		// keep also a list with the index on which withExecution is found, to know on which line to show an error
+		var messagesWithExecution = newLinkedList
```

Comment:
the name of the variable could be updated too

Reply:
Done